src/idl_gen_general.cpp

```diff
@@ -361,6 +361,29 @@ static std::string DestinationValue(const LanguageParameters &lang,
   }
 }
 
+// Cast statements for mutator method parameters.
+// In Java, parameters representing unsigned numbers need to be cast down to their respective type.
+// For example, a long holding an unsigned int value would be cast down to int before being put onto the buffer.
+// In C#, one cast directly cast an Enum to its underlying type, which is essential before putting it onto the buffer.
+static std::string SourceCast(const LanguageParameters &lang,
```

Comment:
DestinationCast sort of does the opposite of what SourceCast does. Basically it upcasts unsigned types so that the correct value will be returned via the accessor. What SourceCast does is the opposite. It downcasts parameters representing unsigned integers before writing them onto the buffer.
I could achieve this with DestinationCast if I added some boolean parameter determining whether or not it should downcast or upcast but I didn't think that would be any better.
Nonetheless, I'm totally open to suggestions on this one.